cmd/skopeo/sync.go

```diff
@@ -166,36 +176,32 @@ func parseRepositoryReference(input string) (reference.Named, error) {
 // destinationReference creates an image reference using the provided transport.
 // It returns a image reference to be used as destination of an image copy and
 // any error encountered.
-func destinationReference(destination string, transport string) (types.ImageReference, error) {
-	var imageTransport types.ImageTransport
+func destinationReference(destination string, destSuffix string, transport string) (types.ImageReference, error) {
 
+	var destRef types.ImageReference
+	var destRefErr error
+
+	destString := path.Join(destination, destSuffix)
 	switch transport {
 	case docker.Transport.Name():
-		destination = fmt.Sprintf("//%s", destination)
-		imageTransport = docker.Transport
+		destRef, destRefErr = docker.Transport.ParseReference(fmt.Sprintf("//%s", destString))
 	case directory.Transport.Name():
-		_, err := os.Stat(destination)
-		if err == nil {
-			return nil, errors.Errorf("Refusing to overwrite destination directory %q", destination)
-		}
-		if !os.IsNotExist(err) {
-			return nil, errors.Wrap(err, "Destination directory could not be used")
-		}
-		// the directory holding the image must be created here
-		if err = os.MkdirAll(destination, 0755); err != nil {
-			return nil, errors.Wrapf(err, "Error creating directory for image %s", destination)
+		if err := checkExistAndMkdir(destString); err != nil {
+			return nil, err
 		}
-		imageTransport = directory.Transport
+		destRef, destRefErr = directory.NewReference(destString)
+	case oci.Transport.Name():
+		destRef, destRefErr = oci.NewReference(destination, destSuffix)
 	default:
 		return nil, errors.Errorf("%q is not a valid destination transport", transport)
 	}
-	logrus.Debugf("Destination for transport %q: %s", transport, destination)
 
-	destRef, err := imageTransport.ParseReference(destination)
-	if err != nil {
-		return nil, errors.Wrapf(err, "Cannot obtain a valid image reference for transport %q and reference %q", imageTransport.Name(), destination)
+	if destRefErr != nil {
+		return nil, errors.Wrapf(destRefErr, "Cannot obtain a valid image reference for transport %q and reference %q", transport, destString)
```

Comment:
`destString` is not actually used on the OCI path … I guess this should list all of `transport, destination, destSuffix` individually..